packages/collabswarm/src/crdt-sync-message.ts

```diff
@@ -17,4 +17,14 @@ export interface CRDTSyncMessage<ChangesType> {
    * implementation.
    */
   changes: { [hash: string]: ChangesType | null };
-}
+
+  readersChanges?: ChangesType;
+
+  writersChanges?: ChangesType;
+
+  /**
+   * Optional document keys list. Only populated while loading and receiving a document
+   * key update (due to the removal of an ACL reader).
+   */
+  keychainChanges?: ChangesType;
```

Comment:
why do we call it keychain instead of just `documentKeys` again? You may have created a new type...hmmm just [re-read the def](https://github.com/collabswarm/collabswarm/blob/16c991b45d750aec39a1863e76b8a5b665579417/packages/
collabswarm/src/keychain.ts#L7)...what do you think about:
_KeychainChange -> documentKeychainChange_ so that _export interface Keychain<documentKeychainChange, DocumentKey>_ . Disadvangage is long variable name, but "Do not worry about saving horizontal space as it is far more important to make your code immediately understandable by a new reader" [[google](Do not worry about saving horizontal space as it is far more important to make your code immediately understandable by a new reader)], advantage: clearer integration of `key` and `document key`, separation from acl keys.

Reply:
That makes sense, I can update that

Reply:
Ah, actually upon second read, I do think that leaving the first type parameter of `Keychain` as `KeychainChange` is probably more clear as it is describing the type of a block of changes made to a `Keychain`. I suppose the fully-namespaced version of the `Keychain` interface would be `CollabswarmDocumentKeychain`. But that also is saying that a keychain is only ever applicable to a CollabswarmDocument.

Reply:
Right, the oject name should match the param. How about dropping the project name part and using  `documentKeychain<documentKeychainChange, DocumentKey> `. It's the reference to the Document that I like because it maps well to the AuthProvider interface parameters `PrivateKey, PublicKey, DocumentKey`.

Ah, you mean you want a generic KeyChain object that can be used elsewhere?

Reply:
Nevermind, I just had a "shower thought" that the keychain is obviously documentKeys because the other chains of keys are ACL lists. I don't want to request changes, just to request changes